central/sensor/service/connection/connection_impl.go

```diff
@@ -591,13 +591,27 @@ func (c *sensorConnection) Run(ctx context.Context, server central.SensorService
 			return errors.Wrapf(err, "unable to get audit log file state sync msg for %q", c.clusterID)
 		}
 
-		// Send the audit log state to sensor even if the the user has it disabled (that's set in dynamic config). When enabled, sensor will use it correctly
+		// Send the audit log state to sensor even if the user has it disabled (that's set in dynamic config). When enabled, sensor will use it correctly
 		if err := server.Send(msg); err != nil {
 			return errors.Wrapf(err, "unable to sync audit log file state to cluster %q", c.clusterID)
 		}
 	}
 
-	metrics.IncrementSensorConnect(c.clusterID, c.sensorHello.GetReconnect())
+	switch c.sensorHello.GetSensorState() {
+	case central.SensorHello_RECONNECT:
+		metrics.IncrementSensorConnect(c.clusterID, true, true)
+	case central.SensorHello_STARTUP:
+		metrics.IncrementSensorConnect(c.clusterID, false, true)
+	case central.SensorHello_UNKNOWN:
+		metrics.IncrementSensorConnect(c.clusterID, false, false)
```

Comment:
With the enum in grpc, we could now replace the two bools of `IncrementSensorConnect` with a string, so that the enum value would be stringified.

Reply:
So you mean having the metric with a single string value as well? Rather than with `"reconnect", "restartPreventionEnabled"`?

Reply:
Ok, I've made the change and I prefer this way also. Check latest commit